v2/pkg/engine/postprocess/postprocess.go

```diff
@@ -161,3 +177,17 @@ func (p *Processor) createFetchTree(res *resolve.GraphQLResponse) {
 		ChildNodes: children,
 	}
 }
+
+// collectDataSourceInfos returns the list of involved data sources of the operation
+func collectDataSourceInfos(node *resolve.FetchTreeNode) (list []resolve.DataSourceInfo) {
+
+	if node.Item != nil && node.Item.Fetch != nil {
+		list = append(list, node.Item.Fetch.DataSourceInfo())
+	}
+
+	for _, childNode := range node.ChildNodes {
+		list = append(list, collectDataSourceInfos(childNode)...)
+	}
+
+	return unique.SliceElements(list)
```

Comment:
could we use https://pkg.go.dev/slices#Compact?

Reply:
Nice, never heard about it.